### PR TITLE
🌐 [i18n-KO] Translated `modular_transformers.md` to Korean

### PR DESCRIPTION
**Pull request by yijun-lee**
# What does this PR do?

Translated the `modular_transformers.md` file of the documentation to Korean.
Thank you in advance for your review.

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [ ] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

May you please review this PR?
@jungnerd, @cjfghk5697, @yijun-lee, @mreraser

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)

@stevhliu May you please review this PR?

### CODE REVIEW

File: docs/source/ko/modular_transformers.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# 모듈식 트랜스포머 [[modular-transformers]]
+
+`transformers`는 의견이 반영된 프레임워크이며, 우리의 철학은 다음의 [개념 가이드](./philosophy)에 정의되어 있습니다.
```
Comment (mreraser) [inline docs/source/ko/modular_transformers.md:3]:
```suggestion
`transformers`는 opinionated(자기 의견이 강한) 프레임워크이며, 우리의 철학은 다음의 [개념 가이드](./philosophy)에 정의되어 있습니다.
```
opinionated 라는 원문의 내용을 살리는 것이 좋을 것 같아 위와 같이 제안드립니다.

File: docs/source/ko/modular_transformers.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+이 철학의 핵심은 라이브러리의 [단일 모델, 단일 파일](https://huggingface.co/blog/transformers-design-philosophy) 측면에서 잘 나타납니다. 이 구성 요소의 단점은 파일 간에 구성 요소의 상속과 임포트 가능성을 제한한다는 것입니다.
+
+그 결과, 모델 구성 요소가 여러 파일에 걸쳐 반복되는 경향이 있습니다. `transformers`에는 모델 수만큼 많은 어텐션 레이어가 정의되어 있으며, 그 중 상당수는 서로 동일합니다. 안타깝게도, 수정과 변경 사항이 코드의 특정 부분에 적용되면서 독립적인 구현들이 서로 분기되는 경향이 있습니다.
+
+이 문제를 적절히 해결하기 위해, 우리는 라이브러리 전체에 "복사본"의 개념을 도입했습니다. 코드가 다른 코드의 복사본임을 나타내는 주석을 추가함으로써, CI 및 로컬 명령을 통해 복사본이 분기되지 않도록 강제할 수 있습니다. 그러나 복잡성이 낮더라도 이는 종종 매우 번거로운 작업입니다.
+
+마지막으로, 이 방식은 우리가 줄이고자 하는 상당한 오버헤드를 모델 기여 과정에 추가하게 됩니다. 이 접근 방식은 종종 모델 기여에 모델링 코드(~1,000줄), 프로세서(~500줄), 테스트, 문서 등을 추가해야 합니다. 모델 기여 PR은 대부분 3,000~5,000줄 이상의 코드를 추가하며, 이 중 많은 부분이 반복되는 문장입니다.
```
Comment (mreraser) [inline docs/source/ko/modular_transformers.md:11]:
```suggestion
마지막으로, 이 방식은 우리가 줄이고자 하는 상당한 오버헤드를 모델 기여 과정에 추가하게 됩니다. 이 접근 방식은 종종 모델 기여에 모델링 코드(~1,000줄), 프로세서(~500줄), 테스트, 문서 등을 추가해야 합니다. 모델 기여 PR은 대부분 3,000~5,000줄 이상의 코드를 추가하며, 이 중 많은 부분이 boilerpate 코드입니다.
```
원문을 살려서 번역해봤습니다.

Comment (cjfghk5697) [reply docs/source/ko/modular_transformers.md:11]: @mreraser boilerpate 오타난 거 같습니다. boilerplate로 수정 가능할까요?!

File: docs/source/ko/modular_transformers.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+## 무엇인가요 [[what-is-it]]
+
+모듈식 트랜스포머는 모델 폴더에 "모듈식" 파일의 개념을 도입합니다. 이 모듈식 파일은 일반적으로 모델링/프로세싱 파일에서 허용되지 않는 코드를 허용하며, 이는 인접한 모델로부터의 임포트와 클래스 간의 상속을 허용합니다.
+
+이 모듈식 파일은 각각의 별도의 모듈에서 정의되었을 모델, 프로세서 및 구성 클래스를 정의합니다.
+
+마지막으로, 이 기능은 모듈식 파일을 "풀어내어" 단일 모델, 단일 파일 디렉토리 구조로 변환하는 새로운 `linter`를 도입합니다. 이 파일들은 스크립트가 실행될 때마다 자동으로 생성되며, 기여해야 할 내용을 모듈식 파일, 그리고 기여된 모델과 다른 모델 간의 차이점으로만 줄여줍니다.
+
+모델 사용자는 단일 파일 인터페이스를 임포트하고 사용하게 되므로, 여기에는 변화가 없을 것입니다. 이를 통해 우리는 양쪽 세계의 장점을 결합하고자 합니다: 간단한 기여를 가능하게 하면서도 우리의 철학을 유지하는 것입니다.
```
Comment (cjfghk5697) [inline docs/source/ko/modular_transformers.md:23]:
```suggestion
모델 사용자는 단일 파일 인터페이스를 임포트하고 사용하게 되므로, 여기에는 변화가 없을 것입니다. 이를 통해 간단한 기여를 가능하게 하면서도 우리의 철학을 유지하는 양쪽의 장점을 결합하고자 합니다.
```
한번 자연스럽게 바꿔봤습니다!

File: docs/source/ko/modular_transformers.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+따라서 이는 `# Copied from` 마커의 대체품이며, 이전에 기여된 모델은 앞으로 몇 달 내에 새로운 모듈식 트랜스포머 형식으로 이동할 것으로 예상됩니다.
```
Comment (cjfghk5697) [inline docs/source/ko/modular_transformers.md:25]:
```suggestion
따라서 이는 `# Copied from` 마커의 대체품이며, 이전에 기여된 모델은 앞으로 몇 달 내에 새로운 모듈식 트랜스포머 형식으로 전환될 예정입니다.
```

File: docs/source/ko/modular_transformers.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+### 자세한 내용 [[details]]
+
+상속을 풀어내고 모듈식 파일로부터 모든 단일 파일을 생성하는 "linter"는 파이썬 사용자에게 보이지 않는 상태로 상속을 평탄화합니다. 현재로서는, linter는 **단일** 수준의 상속을 평탄화합니다.
```
Comment (cjfghk5697) [inline docs/source/ko/modular_transformers.md:29]:
```suggestion
“linter”는 상속 구조를 풀어서 모듈화된 파일로부터 모든 단일 파일을 생성하며, Python 사용자들에게는 그 과정이 보이지 않도록 동작합니다. 현재 linter는 **단일** 수준의 상속만을 평탄화합니다.
```

**Review by stevhliu — APPROVED** [review]
Nice, thanks for your contribution!

File: docs/source/ko/_toctree.yml [hunk 6]
```diff
@@ -137,6 +137,8 @@
     title: 문제 해결
   - local: in_translation
     title: (번역중) Interoperability with GGUF files
+  - local: modular_transformers
+    title: 'transformers'에서의 모듈성
```
Comment (stevhliu) [inline docs/source/ko/_toctree.yml:141]:
```suggestion
    title: `transformers`에서의 모듈성
```

**Review by stevhliu — COMMENTED** [review]
May need to fix this in the toctree title

**Comment by yijun-lee** [comment]
I have resolved it! Thank you :) @stevhliu

File: docs/source/ko/_toctree.yml [hunk 7]
```diff
@@ -137,6 +137,8 @@
     title: 문제 해결
   - local: gguf
     title: GGUF 파일들과의 상호 운용성
+  - local: modular_transformers
+    title: `transformers`에서의 모듈성
```
Comment (stevhliu) [inline docs/source/ko/_toctree.yml:141]: Ah sorry, I think we just can't have any other characters in the title here! 😅 

```suggestion
    title: transformers에서의 모듈성
```

**Pull request merged**